lib/parameter_substitution/expression.rb

```diff
@@ -37,6 +37,14 @@ def substitution_parameter_names
       @expression_list.map_compact(&:parameter_name)
     end
 
+    def method_names
+      @expression_list.map_compact do |expression|
+        expression.try(:method_calls)&.map_compact do |method_call|
+          method_call.name.to_s
+        end
+      end.flatten
```

Comment:
This nested iteration wasn't bad for a proof of concept, but I think we can clean it up a little now that we know the logic is solid.